File: 0134-gas-station/0134-gas-station.java
class Solution {
    public int canCompleteCircuit(int[] gas, int[] cost) {
        int totalGas = 0; 
        int currGas = 0; 
        int candidate = 0; 
        for(int i = 0; i < gas.length; i++){
            totalGas += gas[i] - cost[i]; 
            currGas += gas[i] - cost[i]; 

            if(currGas < 0){
                candidate = i + 1; 
                currGas = 0; 
            }
        }

        return totalGas < 0 ? -1 : candidate; 
    }
}

File: 0134-gas-station/0134-gas-station.py
class Solution:
    def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
        totalGas = 0 
        currGas = 0 
        currLoc = 0
        answer = -1 

        for index in range(0, len(gas)):
            totalGas += gas[index] - cost[index]
            currGas = gas[index] - cost[index] 

            if currGas < 0:
                currLoc = index + 1
            else:
                answer = currLoc 
        
        if totalGas < 0 : return - 1
        return answer
